sdk/cosmosdb/cosmos/src/client/Offer/Offers.ts

```diff
@@ -35,14 +35,15 @@ export class Offers {
    */
   public query<T>(query: SqlQuerySpec, options?: FeedOptions): QueryIterator<T>;
   public query<T>(query: SqlQuerySpec, options?: FeedOptions): QueryIterator<T> {
-    return new QueryIterator(this.clientContext, query, options, (innerOptions) => {
+    return new QueryIterator(this.clientContext, query, options, (diagnosticNode, innerOptions) => {
       return this.clientContext.queryFeed<T>({
         path: "/offers",
         resourceType: ResourceType.offer,
         resourceId: "",
         resultFn: (result) => result.Offers,
         query,
         options: innerOptions,
+        diagnosticNode,
       });
     });
   }
```

Comment:
have we missed attaching diagnostics object while reading offer?

Reply:
Didn't understand this.